core/src/com/etheller/warsmash/parsers/jass/Jass2.java

```diff
@@ -1244,3914 +1039,2266 @@ public boolean call(final CUnit unit) {
+				return BooleanJassValue.TRUE;
 			});
-			jassProgramVisitor.getJassNativeManager().createNative("GetClickedDialog", new JassFunction() {
-				@Override
-				public JassValue call(final List<JassValue> arguments, final GlobalScope globalScope,
-						final TriggerExecutionScope triggerScope) {
-					return new HandleJassValue(dialogType,
-							((CommonTriggerExecutionScope) triggerScope).getClickedDialog());
+			jassProgramVisitor.getJassNativeManager().createNative("UnitItemInSlot", (arguments, globalScope, triggerScope) -> {
+				final CUnit whichUnit = arguments.get(0).visit(ObjectJassValueVisitor.getInstance());
+				final int whichSlot = arguments.get(1).visit(IntegerJassValueVisitor.getInstance());
+				if (whichUnit == null) {
+					return new HandleJassValue(itemType, null);
 				}
+				final CAbilityInventory inventoryData = whichUnit.getInventoryData();
+				if (inventoryData != null) {
+					return new HandleJassValue(itemType, inventoryData.getItemInSlot(whichSlot));
+				}
+				return new HandleJassValue(itemType, null);
 			});
-			jassProgramVisitor.getJassNativeManager().createNative("TriggerRegisterDialogEvent", new JassFunction() {
-				@Override
-				public JassValue call(final List<JassValue> arguments, final GlobalScope globalScope,
-						final TriggerExecutionScope triggerScope) {
-					final Trigger trigger = arguments.get(0).visit(ObjectJassValueVisitor.getInstance());
-					final CScriptDialog dialog = arguments.get(1).visit(ObjectJassValueVisitor.getInstance());
-					if (dialog == null) {
-						return new HandleJassValue(eventType, RemovableTriggerEvent.DO_NOTHING);
-					}
-					return new HandleJassValue(eventType, dialog.addEvent(trigger));
+			jassProgramVisitor.getJassNativeManager().createNative("SetCameraTargetController", (arguments, globalScope, triggerScope) -> {
+				final CUnit whichUnit = nullable(arguments, 0, ObjectJassValueVisitor.getInstance());
+				final float xoffset = arguments.get(1).visit(RealJassValueVisitor.getInstance()).floatValue();
+				final float yoffset = arguments.get(2).visit(RealJassValueVisitor.getInstance()).floatValue();
+				final boolean inheritOrientation = arguments.get(3).visit(BooleanJassValueVisitor.getInstance());
+				meleeUI.getCameraManager().setTargetController(war3MapViewer.getRenderPeer(whichUnit), xoffset, yoffset, inheritOrientation);
+				return null;
+			});
+			jassProgramVisitor.getJassNativeManager().createNative("PanCameraToTimed", (arguments, globalScope, triggerScope) -> {
+				final float x = arguments.get(0).visit(RealJassValueVisitor.getInstance()).floatValue();
+				final float y = arguments.get(1).visit(RealJassValueVisitor.getInstance()).floatValue();
+				meleeUI.getCameraManager().target.x = x;
+				meleeUI.getCameraManager().target.y = y;
+				return null;
+			});
+			jassProgramVisitor.getJassNativeManager().createNative("CameraSetupGetField", (arguments, globalScope, triggerScope) -> {
+				return new RealJassValue(0.0f);
```

Comment:
Prior to the change, `CameraSetupGetField` had a comment saying `TODO NYI` similar to other methods below it. Not sure if we care.